### PR TITLE
Implement Queen Attack generator

### PR DESCRIPTION
**Pull request by jpreese**
One thing to note is that our Example for this one isn't quite complete, so I'm going to start that now.

I'm open to suggestions on the generator for this one.. it feels like a lot, but it could just be a more complicated test to generate. Nothing felt too hacky, and most of it made sense to me.

Edit: Example updated to pass new tests.

### CODE REVIEW

**Comment by ErikSchierboom** [comment]
I'm a bit confused. I would expect the `Create` method to actually create a queen, not return a boolean value.

**Comment by jpreese** [comment]
That was my initial approach as well. I added the Create() method in the stub and had it return type Queen. Though the canonical data has it returning 0 for success, and -1 for failure, I'm not sure why it uses a number for creation, but a boolean for attacking.

Though now that I'm thinking aloud, it may be better to not take the canonical data so literally, and actually go back to Create returning Queen, but throw an exception.

**Comment by ErikSchierboom** [comment]
Agreed! The canonical data can be quite weird sometimes.

**Comment by jpreese** [comment]
Lets give that approach a whirl.

**Comment by jpreese** [comment]
In case the github ping didn't go off for some reason, this has been changed to throw exceptions.

File: exercises/queen-attack/QueenAttackTest.cs [hunk 1]
```diff
@@ -1,55 +1,93 @@
...
 using Xunit;
+using System;
 
 public class QueenAttackTest
 {
     [Fact]
-    public void Cannot_occupy_same_space()
+    public void Queen_with_a_valid_position()
+    {
+        var actual = QueenAttack.Create(2, 2);
```
Comment (ErikSchierboom) [inline exercises/queen-attack/QueenAttackTest.cs]: Perhaps add a comment below this line to indicate that we are testing that no exception is thrown.

File: generators/Exercises/QueenAttack.cs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        }
+
+        protected override string RenderTestMethodBodyAssert(TestMethodBody testMethodBody)
+        {
+            if (testMethodBody.CanonicalDataCase.Property == "canAttack")
+            {
+                return RenderCanAttackAssert(testMethodBody);
+            }
+
+            if(testMethodBody.UseVariableForTested)
```
Comment (ErikSchierboom) [inline generators/Exercises/QueenAttack.cs]: Add space after `if`

File: generators/Exercises/QueenAttack.cs [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+namespace Generators.Exercises
+{
+    public class QueenAttack : Exercise
+    {
+        protected override void UpdateCanonicalData(CanonicalData canonicalData)
+        {
+            foreach(var canonicalDataCase in canonicalData.Cases)
+            {
+                if(canonicalDataCase.Property == "create")
```
Comment (ErikSchierboom) [inline generators/Exercises/QueenAttack.cs]: Add space after `if`

File: generators/Exercises/QueenAttack.cs [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+using System;
+using System.Collections.Generic;
+
+namespace Generators.Exercises
+{
+    public class QueenAttack : Exercise
+    {
+        protected override void UpdateCanonicalData(CanonicalData canonicalData)
+        {
+            foreach(var canonicalDataCase in canonicalData.Cases)
```
Comment (ErikSchierboom) [inline generators/Exercises/QueenAttack.cs]: Add space after `foreach`

**Comment by jpreese** [comment]
I really need to remember to update my visual studio formatting preferences.

**Comment by ErikSchierboom** [comment]
> I really need to remember to update my visual studio formatting preferences.

You really shouldn't have to remember, we should have the tooling do that! I've created an issue for this: #398

**Comment by jpreese** [comment]
I don't think it's worth making a special template for the initial case of making a comment, so I just added it to the test description.

**Comment by ErikSchierboom** [comment]
@jpreese Did you re-run the test generator after making that change? It doesn't appear in the test file :)

**Comment by ErikSchierboom** [comment]
Thanks a lot! 🎉

**Pull request merged**